### PR TITLE
first time code review

### PR DESCRIPTION
**Pull request by qinben10**
first time code review

### CODE REVIEW

**Comment by qinben10** [comment]
database diagrams: 

database relationship on code views:
////customer view:
User.hasMany(Order);
User.hasOne(Cart);
Cart.belongsTo(User);
////order view:
Order.belongsTo(User);
Order.hasMany(Product);
Product.belongsTo(Order);
Order.hasMany(OrderItem);
OrderItem.belongsTo(Order);
Product.hasMany(OrderItem);
OrderItem.belongsTo(Product);
Order.hasOne(Shipment)
Shipment.belongsTo(Order);
Order.hasMany(Payment);
Payment.belongsTo(Order);

File: CODEOWNERS
Comment (Semiroundpizza8) [inline CODEOWNERS]: Any specific reasons why you're adding this to code owners? This file should generally be controlled by us since It allows us to set up behind the scenes stuffs

File: server/main.js
Comment (Semiroundpizza8) [inline server/main.js]: I'd be careful about keeping force:true in here since it'll wipe out your database every time the server starts up - this might cause issues later on. In general, I'd keep force:true limited to just being inside of your seed.

File: client/components/Model.js
Comment (Semiroundpizza8) [inline client/components/Model.js]: It sounds like model might not be the best name for this variable since a model is a property of the object being stored in it - is there something more meaningful we can call this? Or would there perhaps be some cleaner way of getting this information that circumvents storing it in a universal object?

File: client/components/Models.js
Comment (Semiroundpizza8) [inline client/components/Models.js]: Keep console.logs out of production code where you can. This can lead to security vulnerabilities

File: client/components/Models.js
Comment (Semiroundpizza8) [inline client/components/Models.js:11]: While we do want this check to happen (to make sure that we're properly handling what happens when we're out of cars) it looks like it's muddying the rest of the HTML right now, being a little hard to interoperate. It might be worth thinking about if we can move this check somewhere else or rephrase it in a way that'll save us some indentation

File: client/components/SingleModel.js
Comment (Semiroundpizza8) [inline client/components/SingleModel.js]: Since we're just grabbing properties with the same name as their equivalent on state off of the model object here it might be worth thinking about if there's a cleaner way to write this

File: client/components/SingleModel.js
Comment (Semiroundpizza8) [inline client/components/SingleModel.js]: Here's another console.log that should be cut before we get to production

File: client/components/SingleModel.js
Comment (Semiroundpizza8) [inline client/components/SingleModel.js]: Great use of destructuring here!

File: client/store/index.js
Comment (Semiroundpizza8) [inline client/store/index.js]: Is there a reason this exists? I'm worried it might be dangerous setting state to undefined

File: public/index.html
Comment (Semiroundpizza8) [inline public/index.html]: We talked about this in voices but just wanted to log it in here too - be careful about using relative paths in your index.html - it might cause some weird bugs

**Review by Semiroundpizza8 — COMMENTED** [review]
Putting up some comments for the front-end, taking a look at the back-end next

File: CODEOWNERS
Comment (leafoflegend) [reply CODEOWNERS]: Yeah, this file can't be changed - we'll have to block the review until this change is removed.

File: README.md
Comment (leafoflegend) [inline README.md]: 👍

File: client/components/Header.js
Comment (leafoflegend) [inline client/components/Header.js:9]: I enjoy using a `nav` tag!

File: client/components/Routes.js
Comment (leafoflegend) [inline client/components/Routes.js:31]: double padded line here?

File: client/components/Routes.js
Comment (leafoflegend) [inline client/components/Routes.js:48]: Perhaps add a default path here as well.

File: client/components/SingleModel.js
Comment (leafoflegend) [inline client/components/SingleModel.js]: `super` should always receive `props`

File: client/components/SingleModel.js
Comment (leafoflegend) [reply client/components/SingleModel.js]: Though I would say inconsistent linting, if you are using spacing on import destructures, use it on object destructures.

e.g.

```javascript
const { a, b, c } = thing;
```

File: client/store/redux/products.js
Comment (leafoflegend) [inline client/store/redux/products.js]: `try catch` in `async await`

File: seed/seed.js
Comment (leafoflegend) [inline seed/seed.js]: If products aren't dependent on users, you can parallelize their creation.

File: seed/seed.js
Comment (leafoflegend) [inline seed/seed.js]: wow this is fun

File: server/api/users.js
Comment (leafoflegend) [inline server/api/users.js]: We should **never expose user information like this. We are exposing emails/usernames/password without any sort of verification of who is asking for this information. This immediately blocks this code from shipping until resolved.**

File: server/db/index.js
Comment (leafoflegend) [inline server/db/index.js]: What is the difference between a cart and an order?

**Review by leafoflegend — CHANGES_REQUESTED** [review]
You'll need to fix the security concerns I point out in my review. Also, I'd suggest rethinking some of the models and relationships that were decided upon.

File: client/components/SingleModel.js
Comment (Semiroundpizza8) [reply client/components/SingleModel.js]: Great catch - as an extension of this it might be worth talking about normalizing your teams linters if you haven't already. I'd think that most teams consider it best practice to make sure that your linter is run before code is merged in, my team at work runs the linter as part of our CI process and blocks merges until fixes are made.

File: CODEOWNERS
Comment (qinben10) [reply CODEOWNERS]: haha, i added this code when I pushed my first branch to repo....just to test it out I was able to create my branch

File: server/main.js
Comment (00Halfman00) [reply server/main.js]: good to know

File: client/components/SingleModel.js
Comment (00Halfman00) [reply client/components/SingleModel.js]: Half our team doesn't use linters, so they would either have to install one or manually perform this task. lol

File: client/components/Model.js
Comment (00Halfman00) [reply client/components/Model.js]: I see... There seems to be an overuse of this name, since our components are also 'models'. Perhaps auto would be a good replacement.

File: client/store/index.js
Comment (00Halfman00) [reply client/store/index.js]: lol. Fancy code out of nowhere. I myself wonder where this was referenced from.

File: README.md
Comment (00Halfman00) [reply README.md]: I thought it weird to start up an e-commerce app and not have any products immediately available, so I moved products out of the manual seed file. Instead, they are created when the server starts. The 'fake' data is on the seed file though.

File: client/components/Routes.js
Comment (00Halfman00) [reply client/components/Routes.js:48]: Got it done a couple days ago.
<Route path="/" exact component={Home} />

Comment (00Halfman00) [reply client/components/Routes.js:32]: <Route path="/" exact component={Home} />

File: server/main.js
Comment (00Halfman00) [reply server/main.js]: good to know.

File: client/components/Routes.js
Comment (00Halfman00) [reply client/components/Routes.js:31]: Double padded? The name or spacing?

File: client/components/SingleModel.js
Comment (00Halfman00) [reply client/components/SingleModel.js]: Good to know.

File: seed/seed.js
Comment (00Halfman00) [reply seed/seed.js]: lol. I felt the same way when I saw it. lol

File: client/components/SingleModel.js
Comment (leafoflegend) [reply client/components/SingleModel.js]: You can install linters as part of your projects dependencies instead of globally. Then you can implement CI to enforce it (Travis). Theres nothing manual here, just a missing config, some installations, and a lack of CI.

**Comment by qinben10** [comment]
second code review @leafoflegend @Semiroundpizza8

**Pull request closed without merging**
